### PR TITLE
Add mssql support

### PR DESCRIPTION
**Pull request by isthaison**
some thing
-  execsql  add connection mssql
- fix bug duckduckgo-search rate limit
- update typo vi res

### CODE REVIEW

**Review by cike8899 — APPROVED** [review]

**Comment by KevinHuSh** [comment]
**CI failure**.
image

Dockerfile needs refinement. Please refer to [this](https://stackoverflow.com/questions/43984594/automate-installation-of-msodbcsql-with-puppet).

**Comment by isthaison** [comment]
HI @KevinHuSh  It's OK.

File: api/apps/canvas_app.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 from api.utils.api_utils import get_json_result, server_error_response, validate_request, get_data_error_result
 from agent.canvas import Canvas
 from peewee import MySQLDatabase, PostgresqlDatabase
+import pyodbc
```
Comment (KevinHuSh) [inline api/apps/canvas_app.py]:
```suggestion
```

File: api/apps/canvas_app.py [hunk 2]
```diff
@@ -242,8 +242,29 @@ def test_db_connect():
         elif req["db_type"] == 'postgresql':
             db = PostgresqlDatabase(req["database"], user=req["username"], host=req["host"], port=req["port"],
                                     password=req["password"])
-        db.connect()
-        db.close()
+        elif req["db_type"] == 'mssql':
```
Comment (KevinHuSh) [inline api/apps/canvas_app.py:245]:
```suggestion
        elif req["db_type"] == 'mssql':
            import pyodbc
```

**Comment by yuzhichang** [comment]
@isthaison This PR introduced [duckduckgo_search 6.4.1](https://github.com/deedy5/duckduckgo_search) and [primp 0.8.3](https://github.com/deedy5/primp). The author deedy5 just deleted them from github and pypi. We decide to go back to duckduckgo 6.1.9.

**Comment by isthaison** [comment]
I got it 😬😬😬

**Pull request merged**